### ISSUE TITLE
Link "OPEN" option should only be used to display non-duplicate information

### ISSUE THREAD

**Issue by fishytodd** [WCGA/West-Coast-Ocean-Data-Portal#9]
This is related to the fix we implemented in [Ecotrust/wc-data-registry issue #117](https://github.com/Ecotrust/wc-data-registry/issues/117)

We've allowed OPEN to apply to any link in the metadata which is leading to some duplication of links. Technically this is no a bug, but it is not ideal.

Ideally we would like OPEN to apply only after all other types of more specific links have been detected and displayed, and when it is used, it should never duplicate a URL that is already being displayed with another label.

Ecotrust/wc-data-registry#123

**Comment by fishytodd** [comment]
From User Surveys - I feel like the word 'DOWNLOAD' would be more accurate, though the ZIP link is pointing to the same file...

**Comment by tchaddad** [comment]
Same issue as https://github.com/WCGA/West-Coast-Ocean-Data-Portal/issues/13

Moving user suggestion to new issue (as it is not about OPEN)

**Comment by tchaddad** [comment]
I just pushed an improvement to the 'OPEN' label that should prevent it from mislabeling things like 'ZIP' or 'KML' as 'OPEN'.

Under the new logic, 'OPEN' will be more applied to items identified as website, rather than service URLs.

So far so good. @emiliom I'd like you to poke around and see if you like the new behavior (compare DEV to PROD to see the difference - the fix is on DEV).

**Comment by tchaddad** [comment]
Allison mentions an example where this fix is not working as intended: https://github.com/WCGA/West-Coast-Ocean-Data-Portal/issues/31#issuecomment-76102528 ... will follow up, using these records to test:

http://wcga-vm01.sdsc.edu/discover/#?s=Sound%20GIS%20Test&text=surfrider

**Comment by emiliom** [comment]
Thanks for all your hard work, Tanya!! I'm reporting on the net effect on NANOOS records. First, for reference, we have 5 production records in the NANOOS source, and one new iso-1 record added within the last 24 hours, that's only on staging. 4 of those production records are largely identical and generated by the same process so they will respond the same (ideally!) to all changes. I'll call them "OSU Climatology"
1. The OSU Climatology records previously had an Open link type (yay), but it pointed to an undesirable link, the WMS GetCapabilities (boo). On the staging portal, the Open link is still there (yay) and it points to a more desirable url (yay); I can't confirm yet if it's the best url we think it should be pointing to, based on what's on the iso-2 records. I'll try to confirm that tomorrow or early next week. Also, as I just reported on [issue #23](https://github.com/WCGA/West-Coast-Ocean-Data-Portal/issues/23#issuecomment-76247760), the WMS link type is now exposed (yay). So, all good.
2. The [ROMS record](http://wcga-vm01.sdsc.edu/discover/#?s=NANOOS&text=roms) lost the Open link type it correctly had (you can see it on the production portal). FYI, this is an iso-1 record.
3. The [new NVS record](http://wcga-vm01.sdsc.edu/discover/#?s=NANOOS&text=nvs) has no Open link type. This is the iso-1 record added yesterday and currently only on staging. It points to an application (a url), and the Open link type is really the only link it has for now, so the record is kind of worthless w/o it. I can't report on a "before" behavior, b/c it got harvested and index by solr for the first time roughly at the same time as your changes were being applied. But I can confirm that the link is correctly assigned to the "Open" label in the Geoportal client.

**Comment by tchaddad** [comment]
Thanks Emilio - glad that item 1 is working how you expect.

Items 2 and 3 that you mention are the same condition/case reported in my comment directly above (from Allison).

I was thinking about this case/fix on my way home from work yesterday and I think I have a solution. I need to do a little research today, and hope to push a refinement soon.

**Comment by emiliom** [comment]
@tchaddad, I was wondering where this stands. I have a [NANOOS record for NVS](http://wcga-vm01.sdsc.edu/discover/#?text=nvs) that I started up but haven't finished (that's my item 3 above), partly b/c I was waiting to see how this issue shook out (ok, that's just a lame excuse to set this aside and work on other things ... sigh). Maybe I should just finish up the record, then ask Todd to push it to live, and not worry about this issue for this case, given that the Open link type is probably recognized just fine with the live code?
